File: rc-comercial-fe/src/utils/uuid7.js
// UUID v7 (ordenado por tiempo), espejo de Domain/Common/Uuid7.cs — necesario
// para generar el Id de la venta en el cliente (idempotencia / futuro offline).
export function uuid7() {
  const bytes = new Uint8Array(16)
  crypto.getRandomValues(bytes)

  const ms = BigInt(Date.now())
  bytes[0] = Number((ms >> 40n) & 0xffn)
  bytes[1] = Number((ms >> 32n) & 0xffn)
  bytes[2] = Number((ms >> 24n) & 0xffn)
  bytes[3] = Number((ms >> 16n) & 0xffn)
  bytes[4] = Number((ms >> 8n) & 0xffn)
  bytes[5] = Number(ms & 0xffn)

  bytes[6] = (bytes[6] & 0x0f) | 0x70 // versión 7
  bytes[8] = (bytes[8] & 0x3f) | 0x80 // variante RFC 4122

  const hex = [...bytes].map((b) => b.toString(16).padStart(2, '0')).join('')
  return `${hex.slice(0, 8)}-${hex.slice(8, 12)}-${hex.slice(12, 16)}-${hex.slice(16, 20)}-${hex.slice(20)}`
}

File: src/RcComercial.Domain/Common/Uuid7.cs
using System.Security.Cryptography;

namespace RcComercial.Domain.Common;

/// <summary>
/// Generador de UUID v7 (ordenado por tiempo) para .NET 8.
/// Mantiene los índices PK sin fragmentación y permite generar IDs
/// en el cliente (modo offline) sin colisiones.
/// </summary>
public static class Uuid7
{
    public static Guid NewGuid()
    {
        Span<byte> bytes = stackalloc byte[16];
        RandomNumberGenerator.Fill(bytes);

        long ms = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        bytes[0] = (byte)(ms >> 40);
        bytes[1] = (byte)(ms >> 32);
        bytes[2] = (byte)(ms >> 24);
        bytes[3] = (byte)(ms >> 16);
        bytes[4] = (byte)(ms >> 8);
        bytes[5] = (byte)ms;

        bytes[6] = (byte)((bytes[6] & 0x0F) | 0x70); // versión 7
        bytes[8] = (byte)((bytes[8] & 0x3F) | 0x80); // variante RFC 4122

        return new Guid(bytes, bigEndian: true);
    }
}
